fs2/src/main/scala/log/effect/fs2/LogSelector.scala

```diff
@@ -0,0 +1,41 @@
+package log.effect
+package fs2
+
+import cats.Applicative
+
+/**
+  * It helps adding logging capability to components
+  * allowing them to use a `LogWriter[F]` if in scope
+  * or defaulting to noOpLog if none is provided.
+  *
+  * Example:
+  *
+  * def create[F[_]: LogSelector](a: String): ALoggingClient[F] =
+  *   new ALoggingClient[F] {
+  *     def do = LogSelector[F].writer.info("something")
+  *   }
+  *
+  * def useTheClient[F[_]](address: String): F[Unit] = {
+  *   implicit val logger: LogWriter[F] = consoleLog[F]
+  *   create[F](address)
+  * }
+  * will log using `logger`
+  *
+  * def useTheClient[F[_]](address: String): F[Unit] = create[F](address)
+  * will not log
+  */
+final class LogSelector[F[_]](val writer: LogWriter[F]) extends AnyVal
```

Comment:
what about calling this `log` instead of `writer`?  I'm thinking about all the potential saved typing in the future :)